app/src/main/kotlin/org/eclipse/kuksa/companion/feature/home/view/AdaptiveAppScreen.kt

```diff
@@ -86,67 +87,86 @@ fun AdaptiveAppScreen(
+        val navigationViewPaddingValues = AdaptiveNavigationView.calculatePaddingValues(windowSizeClass)
+        if (connectionStatusViewModel.connectionState != ConnectionState.CONNECTED) {
+            AdaptiveConnectionStatusView(
+                connectionStatusViewModel,
+                windowSizeClass,
+                Modifier
+                    .zIndex(ZINDEX_OVERLAY)
+                    .padding(navigationViewPaddingValues),
+            )
         }
-        AdaptiveSheetView(
+
+        val connectionStatusViewPaddingValues =
+            ConnectionStatusView.calculatePaddingValues(connectionStatusViewModel, windowSizeClass)
+        AdaptiveColumnRow(
             windowSizeClass = windowSizeClass,
-            modifier = Modifier.fillMaxSize(),
-            isSheetEnabled = selectedPage.isSheetEnabled,
-            sheetContent = {
-                when (selectedPage) {
-                    NavigationPage.DOORS -> DoorControlView(doorControlViewModel)
-                    NavigationPage.TEMPERATURE -> TemperatureControlView(temperatureViewModel)
-                    NavigationPage.LIGHT -> LightControlView(lightControlViewModel)
-                    NavigationPage.WHEELS,
-                    NavigationPage.SETTINGS,
-                    -> { }
-                }
-            },
+            modifier = modifier.fillMaxSize(),
         ) {
-            RamsesView(
-                callback = callback,
+            AdaptiveNavigationView(navigationViewModel, windowSizeClass) { page ->
+                selectedPage = page
+            }
+            AdaptiveSheetView(
+                windowSizeClass = windowSizeClass,
                 modifier = Modifier
-                    .zIndex(ZINDEX_RAMSES_VIEW)
                     .fillMaxSize(),
-            )
+                isSheetEnabled = selectedPage.isSheetEnabled,
+                sheetContent = {
+                    when (selectedPage) {
+                        NavigationPage.DOORS -> DoorControlView(doorControlViewModel)
+                        NavigationPage.TEMPERATURE -> TemperatureControlView(temperatureViewModel)
+                        NavigationPage.LIGHT -> LightControlView(lightControlViewModel)
+                        NavigationPage.WHEELS,
+                        NavigationPage.SETTINGS,
+                        -> {
+                        }
+                    }
+                },
+            ) {
+                RamsesView(
+                    callback = callback,
+                    modifier = Modifier
+                        .zIndex(ZINDEX_CONTENT)
+                        .fillMaxSize(),
+                )
 
-            val overlayModifier = Modifier
-                .zIndex(ZINDEX_OVERLAY)
-                .fillMaxSize()
-                .padding(it)
+                val overlayModifier = Modifier
+                    .zIndex(ZINDEX_OVERLAY)
+                    .fillMaxSize()
+                    .padding(connectionStatusViewPaddingValues)
```

Comment:
pushes the content of the overlay below the ConnectionStatusView (important for settings!)